gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -69,6 +80,18 @@ def update_buildfiles_for_dependency(buildfiles:, dependency:)
 
           buildfile = files.find { |f| f.name == new_req.fetch(:file) }
 
+          # Exception raised to handle issue that arises when buildfiles function (see this file)
+          # removes the build file that contains the dependency itself. So no build file exists to
+          # update dependency, This behaviour is evident for extremely small number of users
+          # that have added separate repos as sub-modules in parent projects
+
+          if buildfile.nil?
```

Comment:
can we have test coverage too please?

Reply:
Should be straightforward to add an example manifest to test, if you run into problems just ask.

Reply:
Added test case to mock non existent dependency file. Please check and let know if that suffices.